mobile/src/user/UserPages.tsx

```diff
@@ -0,0 +1,61 @@
+import { FormEvent, useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { Button, FormControl, TextField } from "@mui/material";
+
+import { useUserRepository, useUser } from "../shared/SharedHooks";
+
+export function SignIn() {
+  const navigate = useNavigate();
+  const userRepository = useUserRepository();
+  const [_, setUser] = useUser();
+
+  const [name, setName] = useState<string>();
+
+  const handleSubmit = async function (e: FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+
+    if (!name) {
+      return;
+    }
+
+    try {
+      const signedInUser = await userRepository.signIn(name);
+      setUser(signedInUser);
+      navigate("/");
+    } catch (e) {
+      console.error(e);
+    }
+  };
+
+  return (
+    <form onSubmit={handleSubmit} method="POST">
+      <FormControl>
+        <TextField
+          id="filled-basic"
+          label="Username"
+          variant="filled"
+          onChange={({ target: { value } }) => setName(value)}
```

Comment:
```suggestion
          onChange={event => setName(event.target.value)}
```

is more readable IMHO